Mathlib/Analysis/NormedSpace/DualNumber.lean

```diff
@@ -27,18 +27,18 @@ open TrivSqZeroExt
 
 variable (𝕜 : Type*) {R : Type*}
 
-variable [IsROrC 𝕜] [NormedCommRing R] [NormedAlgebra 𝕜 R]
+variable [IsROrC 𝕜] [NormedCommRing R] [Algebra ℚ R] [NormedAlgebra 𝕜 R]
```

Comment:
Shouldn't `Algebra ℚ R` follow from `NormedAlgebra 𝕜 R`?

Reply:
Unfortunately that creates instance diamonds, so is a bad idea. Yes, any concrete `R` should satisfy the former if it satisfies the latter, but that doesn't mean we can safely perform the construction in a theorem *statement*.

Reply:
Yep, that makes sense.

Reply:
Though in this file it turns out that `𝕜` is no longer needed at all, thanks to #9491 landing in the meantime.